### PR TITLE
aaron/APPEALS-8945

### PR DESCRIPTION
**Pull request by Aaron-Willis**
Resolves #APPEALS-8945

### Description
Added two jobs (one for legacy and ama appeals) that will do a single migration of all open and cancelled status appeals into the appeal states table.

### Acceptance Criteria
- [x] Add logic to fetch_all_active_ama_appeals & fetch_all_active_legacy_appeals that will insert records into the Appeal States table:
- [x] Added to the async job whitelist ***TO DO***
- [x] Has a perform method
- [x] Has error handling:
- [x] If an exception is encountered during processing log and continue job
Logic exists to query all open AMA and Legacy appeals for processing
- [x] Method outlines exists for the following methods:
- [x] map_appeal_ihp_state
param: appeal
- [x] map_appeal_privacy_act_state
param: appeal
- [x] map_appeal_hearing_scheduled_state
param: appeal
- [x] map_appeal_hearing_postponed_state
param: appeal
- [x] map_appeal_hearing_withdrawal_state
param: appeal
- [x] map_appeal_hearing_scheduled_in_error_state
param: appeal
- [x] map_appeal_cancelled_state
param: appeal
- [x] map_appeal_docketed_state
param: appeal
- [x] map_appeal_decision_mailed_state
param: appeal

### CODE REVIEW

File: app/jobs/fetch_all_active_ama_appeals_job.rb [hunk 1]
```diff
@@ -1,33 +1,111 @@
...
+  #
+  # Params: None
+  #
+  # Returns: nil
+  def find_and_create_appeal_state_for_active_ama_appeals
+    Task.where(
+      type: "RootTask",
+      appeal_type: "Appeal",
+      status: Task.open_statuses.concat([Constants.TASK_STATUSES.cancelled])
+    ).find_in_batches(batch_size: 1_000) do |root_tasks|
```
Comment (jshields13) [inline app/jobs/fetch_all_active_ama_appeals_job.rb:31]: Lets move the batch size to an environment variable

Comment (jshields13) [reply app/jobs/fetch_all_active_ama_appeals_job.rb:31]: ENV["STATE_MIGRATION_JOB_BATCH_SIZE"]

Comment (jshields13) [reply app/jobs/fetch_all_active_ama_appeals_job.rb:31]: Can default the environments

File: spec/jobs/fetch_all_active_ama_appeals_job_spec.rb [hunk 2]
```diff
@@ -1,79 +1,70 @@
 # frozen_string_literal: true
 
-describe FetchAllActiveAmaAppealsJob, type: :job do
+describe FetchAllActiveLegacyAppealsJob, type: :job do
```
Comment (jshields13) [inline spec/jobs/fetch_all_active_ama_appeals_job_spec.rb:3]: Name does not match with spec name

**Review by jshields13 — CHANGES_REQUESTED** [review]

**Review by jshields13 — APPROVED** [review]

File: spec/jobs/fetch_all_active_ama_appeals_job_spec.rb (hunk 2 above)
Comment (Aaron-Willis) [reply spec/jobs/fetch_all_active_ama_appeals_job_spec.rb]: resolved

File: app/jobs/fetch_all_active_ama_appeals_job.rb (hunk 1 above)
Comment (Aaron-Willis) [reply app/jobs/fetch_all_active_ama_appeals_job.rb]: resolved

**Pull request merged**